File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <iostream>
#include <stack>
using namespace std;

int main()
{

        char str[100001];
        int cnt = 0;
        scanf("%s", str);
        stack<int> dat;
        for (int i = 0 ; str[i] ; i++) {
                if (str[i] == '(')
                        dat.push(str[i]);
                else if (str[i] == ')' && str[i - 1] == '(') {
                        dat.pop();
                        cnt += dat.size();
                }
                else {
                        cnt++;
                        dat.pop();
                }
        }
        printf("%d\n", cnt);
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] tokens = br.readLine().split("");
        Stack<String> stack = new Stack<>();

        int cnt = 0;
        for (int i = 0; i < tokens.length; i++) {
            if (tokens[i].equals("(")) {
                stack.push(tokens[i]);
            } else if (tokens[i].equals(")") && tokens[i - 1].equals("(")) {
                stack.pop();
                cnt += stack.size();
            } else {
                cnt++;
                stack.pop();
            }
        }
        System.out.println(cnt);
    }
}